File: Array/SlidingWindow/longestOnes.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        i = 0
        left = 0
        res = 0
        n = len(nums)
        for right in range(n):
            if nums[right] == 0:
                k -= 1
            while k < 0:
                if nums[left] == 0:
                    k += 1
                left += 1
            res = max(right - left + 1, res)
        return res

File: Array/SlidingWindow/longestOnes.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int res = 0; int left = 0;
        for (int right = 0; right < nums.size(); ++right) {
            if (!nums[right]) k --;
            while (k < 0) {
                if (!nums[left]) k ++;
                left ++;
            }
            res = max(res, right - left + 1);
        }
        return res;
    }
};